integration/src/main/java/com/aws/greengrass/mqtt/moquette/BrokerKeyStore.java

```diff
@@ -75,8 +78,15 @@ public void initialize() throws KeyStoreException {
      * @throws KeyStoreException If unable to set key entry.
      */
     public void updateServerCertificate(CertificateUpdateEvent certificateUpdate) throws KeyStoreException {
-        // TODO: Support certificate chains
-        Certificate[] certChain = {certificateUpdate.getCertificate()};
+        List<X509Certificate> certChainList = new ArrayList<>();
+        // Add server cert as first entry to the certificate chain
+        certChainList.add(certificateUpdate.getCertificate());
+        // Add rest of the CA certificate chain
```

Comment:
The RFC allows us to omit the self signed CA, which is what we'll want to do here. Normally this wouldn't be a big deal, but we want to try to optimize for size since we have FreeRTOS clients who are concerned with KBs of memory.
I'd like to not make assumptions about whether or not CDA gives us the root. For example, the customer may not provide it, in which case CDA can't give it to us. I'd like us to have some logic here to be flexible.